File: 4_3CNF/Java/Main.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.FileInputStream;
import java.io.IOException;

public class Main {
  public static void main(String args[]) throws IOException{
    // Reads from file
    if(args.length == 0){
      System.out.println("Usare: 'java Main test'");
      return;
    }
    FileInputStream input_stream = new FileInputStream(args[0]);
    CharStream input = CharStreams.fromStream(input_stream); 
    
    // Creates the Lexer
    CNF3Lexer lexer = new CNF3Lexer(input);
    CommonTokenStream tokens = new CommonTokenStream(lexer); 

    // Print the tokens readed
    tokens.fill();
    for(Token token : tokens.getTokens()){
        System.out.println(token.toString());
    }

    // Create the parser
    CNF3Parser parser = new CNF3Parser(tokens);
    ParseTree tree = parser.parse_all();

    if(parser.getNumberOfSyntaxErrors() == 0){
      System.out.println("Parsing result = SUCCESS");
      // Print the parsing tree
      System.out.println(tree.toStringTree(parser));
    }
  }
}

File: 4_3CNF/JavaScript/main.js
var antlr4 = require('antlr4');
var CNF3Lexer = require('./CNF3Lexer').CNF3Lexer;
var CNF3Parser = require('./CNF3Parser').CNF3Parser;

var input = "(b || b || !a)"
var chars = new antlr4.InputStream(input);

// Lexer step
var lexer = new CNF3Lexer(chars);
var tokens  = new antlr4.CommonTokenStream(lexer);

// Print the tokens readed
tokens.fill();
tokens.getTokens(0, input.length).forEach(token =>
    console.log(token.toString())
)

// Parser step
var parser = new CNF3Parser(tokens);
parser.buildParseTrees = true;
var tree = parser.parse_all();

// Print the parsing tree
if(tree.parser._syntaxErrors == 0){
    console.log("Parsing result = SUCCESS")
    console.log(tree.toStringTree(null, parser))
}
